lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -226,7 +226,10 @@ def call(
         ) -> requests.Response:
             """Performs an HTTP request."""
             if remaining_retries < 0:
-                if return_failed_resp and error_response:
+                if not error_response:
+                    raise OpenSearchHttpError()
+
+                if return_failed_resp:
                     return error_response
```

Comment:
what's the difference between this and how it was previously? could we perhaps revert?

Reply:
the `OpenSearchHttpError`raised in line 235 relies on `error_response.text` - if `error_response` doesn't exist, raising that error fails and everything explodes.